### ISSUE TITLE
Unable to join 1.8 server with 1.9+ client when Pledge is enabled

### ISSUE THREAD

**Issue by TypicalFin** [ThomasOM/Pledge#9]
Server version: 1.8.8
Plugins: ViaVersion (4.7.0), PacketEventsTest

When using the PacketEvents test plugin (https://gist.github.com/ThomasOM/e25ec8605871933abbc848b94f4f9f6e), 1.9+ players get kicked immediately when trying to join a 1.8 server (with ViaVersion). I compiled the plugin with the latest snapshot of PacketEvents 2.0 but the issue was also present when using PacketEvents 1.8.4.

Using a 1.8 client works just fine, but 1.9+ versions result in an immediate kick.

Console output:
```
[20:53:29 INFO]: UUID of player TypicalFin is a6560d17-a5d5-3e5a-9d41-ab7942e0f228
[20:53:29 INFO]: TypicalFin[/127.0.0.1:55164] logged in with entity id 149 at ([world]204.84593849966896, 62.0, 177.20572391437955)
[20:53:29 INFO]: activate handler
[20:53:29 INFO]: create frame
[20:53:29 INFO]: server teleport
[20:53:29 INFO]: TypicalFin lost connection: Internal Exception: java.io.IOException: An established connection was aborted by the software in your host machine
[20:53:29 INFO]: TypicalFin left the game.
```

Kick message:
![image](https://github.com/ThomasOM/Pledge/assets/45766416/9217bf75-02a7-4a97-8834-f827110ea498)

**Comment by ThomasOM** [comment]
Commit 8185c6a solved this issue, please verify using Pledge 2.6, then I can close this issue.

**Comment by TypicalFin** [comment]
I no longer get kicked but the plugin seems to be broken on 1.17 and above. I get an INCOMPLETE_FRAME error every time I get teleported:

![image](https://github.com/ThomasOM/Pledge/assets/45766416/bca17dfd-9ad5-43b4-8ddb-1a1edf1051d5)

I assume it has something to do with ping/pong packets as versions below 1.17 work perfectly.
Tested this with PacketEvents 1.8.4 and 2.0.

Here's the stacktrace:
```
[11:30:55 WARN]: A task raised an exception.
java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
        at dev.thomazz.pledge.PlayerHandler.lambda$tickEnd$1(PlayerHandler.java:134) ~[?:?]
        at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:380) [patched_1.8.8.jar:git-PaperSpigot-445]
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:357) [patched_1.8.8.jar:git-PaperSpigot-445]
        at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:116) [patched_1.8.8.jar:git-PaperSpigot-445]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_231]
Caused by: java.lang.reflect.InvocationTargetException
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_231]
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_231]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_231]
        at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_231]
        at dev.thomazz.pledge.util.ChannelHelper.encodeAndCompress(ChannelHelper.java:29) ~[?:?]
        at dev.thomazz.pledge.network.PacketFrameOutboundTailHandler.drain(PacketFrameOutboundTailHandler.java:63) ~[?:?]
        at dev.thomazz.pledge.PlayerHandler.lambda$tickEnd$1(PlayerHandler.java:132) ~[?:?]
        ... 4 more
Caused by: com.viaversion.viaversion.exception.CancelEncoderException$1: This packet is supposed to be cancelled; If you have debug enabled, you can ignore these
```

**Comment by ThomasOM** [comment]
Experiencing the same issue, will update once I have found the cause

**Comment by ThomasOM** [comment]
This should be fixed in 003d91e
The issue was caused by viaversion cancelling the original packet and resending a new one through the channel instead of transforming it completely. As a side effect of this update the transaction packets from pledge should be visible by packet listener plugins.
